workbench-core/infrastructure/src/workbenchCognito.ts

```diff
@@ -105,7 +113,7 @@ export class WorkbenchCognito extends Construct {
       removalPolicy: props.removalPolicy
     };
 
-    const userPoolProps = merge(userPoolDefaults, tempUserPoolProps);
+    const userPoolProps = merge(cloneDeep(userPoolDefaults), tempUserPoolProps);
```

Comment:
I think a better solution would be to do `const userPoolProps = merge({}, userPoolDefaults, tempUserPoolProps);` so we can avoid a potentially expensive `cloneDeep`